Utils/Game.py

```diff
@@ -1,13 +1,337 @@
+        elif self.gb.check_black:
+            r, c = self.gb.black_king_position
+        else:
+            return
+        move_surface = pygame.Surface((self.SQUARE_SIZE, self.SQUARE_SIZE), pygame.SRCALPHA)
+        move_surface.fill(self.DARK_RED)
+        self.screen.blit(move_surface, (self.X_OFFSET + c * self.SQUARE_SIZE, self.Y_OFFSET + r * self.SQUARE_SIZE))
+
+    def show_message(self, color, message):
+        if message is None:
+            message = color[0].upper() + color[1:] + ' won by checkmate'
+        font = pygame.font.Font(None, 72)
+        text = font.render(message, True, self.GREEN)
+        text_rect = text.get_rect(center=(
+            self.X_OFFSET + self.BOARD_SIZE // 2, self.Y_OFFSET + self.BOARD_SIZE // 2))
+        self.screen.blit(text, text_rect)
+
+    def show_promotion(self, position):
+        r, c = position
+        color, positions = ('b', [2, 3, 4, 5]) if r == 7 else ('w', [5, 4, 3, 2])
+        r = -1.1 if r == 0 else 8.1
+        rectangles = [
+            pygame.Rect(self.X_OFFSET + c * self.SQUARE_SIZE, self.Y_OFFSET + r * self.SQUARE_SIZE, self.SQUARE_SIZE,
+                        self.SQUARE_SIZE) for c in positions]
+        identifiers = [color + x for x in ['Q', 'R', 'B', 'N']]
+        self.promoting_pieces = []
+        for rectangle, identifier in zip(rectangles, identifiers):
+            self.screen.blit(self.IMAGES[identifier], rectangle)
+            self.promoting_pieces.append((identifier, rectangle))
+
+    def handle_promotion(self, x, y):
+        for identifier, position in self.promoting_pieces:
+            if position.collidepoint(x, y):
+                self.gb.change_piece(identifier)
+                return
+
+    def select_piece(self, x, y):
+        if 0 < x - self.X_OFFSET < self.BOARD_SIZE and 0 < y - self.Y_OFFSET < self.BOARD_SIZE:
+            row, col = (x - self.X_OFFSET) // self.SQUARE_SIZE, (y - self.Y_OFFSET) // self.SQUARE_SIZE
+            if self.gb.board[col][row] and self.gb.board[col][row].color == self.gb.curr_player:
+                self.selected_piece = self.gb.board[col][row]
+                self.original_pos = (col, row)
+                self.mouse_down = True
+
+    def drag_piece(self, x, y):
+        if 0 < x - self.X_OFFSET < self.BOARD_SIZE and 0 < y - self.Y_OFFSET < self.BOARD_SIZE:
+            self.screen.blit(self.IMAGES[self.selected_piece.get_identificator()],
+                             (x - self.SQUARE_SIZE // 2, y - self.SQUARE_SIZE // 2))
+
+    def update_possible_moves(self):
+        if self.selected_piece:
+            self.moves, self.attack_moves = self.selected_piece.can_move(self.gb)
+            if type(self.selected_piece) == King:
+                self.moves, self.attack_moves = self.gb.castle(self.moves, self.attack_moves, self.selected_piece)
+            self.moves, self.attack_moves = self.gb.preventer(self.moves, self.attack_moves, self.selected_piece)
+
+    def make_move(self, x, y):
```

Comment:
I feel like this method should be broken into smaller chunks of logic, that perform next steps of `make_move`